x-pack/plugins/apm/server/lib/helpers/create_es_client/create_infra_metrics_client/create_infra_metrics_client.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ESSearchRequest, InferSearchResponseOf } from '@kbn/es-types';
+import { APMRouteHandlerResources } from '../../../../routes/typings';
+import { getInfraMetricIndices } from '../../get_infra_metric_indices';
+
+type InfraMetricsSearchParams = Omit<ESSearchRequest, 'index'>;
```

Comment:
@MiriamAparicio I just notice that we forgot to add the required fields we also have for APM client:

```ts
type InfraMetricsSearchParams = Omit<ESSearchRequest, 'index'> & body: {
  size: number;
  track_total_hits: boolean | number;
};
```

This will ensure we always specify `size` and `track_total_hits`.

Reply:
Added required fields here https://github.com/elastic/kibana/pull/142569